### PR TITLE
Hang 10: migrate iOSDeviceManager to Blue Ocean pipeline

### PR DESCRIPTION
**Pull request by AndreyMitsyk**
**Changes:**
- Migrated iOSDeviceManager repository to Jenkins Blue Ocean pipeline.
- Changed test scripts to prevent silent build/tests failings when xcpretty is not installed.
- Enabled notifications and nightly builds.
- Enabled test output in Jenkins.
- Fixed README file.

### CODE REVIEW

**Comment by AndreyMitsyk** [comment]
Jenkins test this please

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+String cron_string = BRANCH_NAME == "develop" ? "H H(0-8) * * *" : ""
```
Comment (jmoody) [inline Jenkinsfile]: Should we add a shebang?

```
#!/usr/bin/env groovy
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](http://calabash-ci.xyz:8081/job/Calabash-iOSDeviceManager/job/develop/badge/icon)](http://calabash-ci.xyz:8081/job/Calabash-iOSDeviceManager/job/develop/)
+
```
Comment (jmoody) [inline README.md]: This link does not work.

I have not found an "embeddable status" URL for the Blue Ocean pipelines.

File: bin/make/build.sh [hunk 3]
```diff
@@ -9,12 +9,10 @@ BUILD_DIR="build"
 XC_PROJECT="iOSDeviceManager.xcodeproj"
 XC_TARGET="iOSDeviceManager"
 
-hash xcpretty 2>/dev/null
-if [ $? -eq 0 ] && [ "${XCPRETTY}" != "0" ]; then
-  XC_PIPE='xcpretty -c'
-else
+XC_PIPE="xcpretty -c"
+hash "xcpretty" 2>/dev/null && [ "${XCPRETTY}" != "0" ] || {
```
Comment (jmoody) [inline bin/make/build.sh]: Interesting rewrite.  I had to test it because I did not understand the last `||`

File: bin/test/ci.sh [hunk 4]
```diff
@@ -48,13 +48,10 @@ make tests
...
-pkill Simulator
-
 if [ "${EXIT_STATUS}" = "0" ]; then
   echo "Tests passed"
   exit 0
 else
   echo "Tests failed."
   exit 1
-fi
+fi
```
Comment (jmoody) [inline bin/test/ci.sh]: Remove all the calabash codesign stuff:  line 17 - 34

Also remove line 36 - 38.

File: bin/test/ci.sh [hunk 5]
```diff
@@ -48,13 +48,10 @@ make tests
 
 EXIT_STATUS=$?
 
-pkill iOSDeviceManager
-pkill Simulator
```
Comment (jmoody) [inline bin/test/ci.sh]: I see these have been added to the `post`.

They also need to be run before starting the other tests.

Maybe add a "setup" node where iOSDeviceManager and Simulator are terminated?

File: README.md [hunk 6]
```diff
@@ -32,9 +34,12 @@ to see usage information.
 ### Testing
 
 ```shell
-$ make test-unit
-$ make test-integration
-$ make test-run-loop
+$ make unit-tests
+$ make integration-tests
+$ make cli-tests
+$ make run-loop-tests
```
Comment (jmoody) [inline README.md]: Remove the `run-loop-tests` line.

File: README.md (hunk 2 above)
Comment (AndreyMitsyk) [reply README.md]: True. This will work when changes will be merged to develop. This will trigger creation of pipeline and image will be available

**Comment by jmoody** [comment]
To debug the erase simulator test add this `before` block to 

https://github.com/calabash/iOSDeviceManager/blob/develop/spec/erase_simulator_spec.rb

```
describe "erase-simulator" do

  before do
    allow(RunLoop::Environment).to receive(:debug?).and_return(true)
  end
  
  let(:device) { IDM::Resources.instance.default_simulator }
```

File: Jenkinsfile (hunk 1 above)
Comment (AndreyMitsyk) [reply Jenkinsfile]: This is not required, but let's add this

File: bin/test/ci.sh (hunk 5 above)
Comment (AndreyMitsyk) [reply bin/test/ci.sh]: Yeah, fair enough.

File: README.md (hunk 2 above)
Comment (AndreyMitsyk) [reply README.md]: How it looks for current branch:
http://calabash-ci.xyz:8081/job/Calabash-iOSDeviceManager/job/feature%252Fjenkins_pipline/badge/icon

**Comment by jmoody** [comment]
I created tasks to track the CI failures.  You can merge this whenever you want @AndreyMitsyk.  Thanks.

**Comment by AndreyMitsyk** [comment]
jenkins retest this please

**Pull request merged**